### PR TITLE
i18n all the things

### PR DESCRIPTION
**Pull request by albemarle**
This should be it for the hard-coded strings.

To be done in a followup:
* Country names
* Error strings

### CODE REVIEW

File: src/components/ProfilePage/AgeMessage.jsx [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+        description="error message"
+        tagName="p"
+      />
+      <a href={accountURL}>
+        <FormattedMessage
+          id="profile.age.set.date"
+          defaultMessage="Set your date of birth"
+          description="label on a link to set birthday"
+        />
+      </a>
```
Comment (abutterworth) [inline src/components/ProfilePage/AgeMessage.jsx]: Oh boy this makes our markup hard to read :(

Comment (albemarle) [reply src/components/ProfilePage/AgeMessage.jsx]: If you get sick of it, you can pull it out to a .messages.jsx file... maybe we'll standardize on something later.

File: src/components/ProfilePage/Name.jsx [hunk 2]
```diff
@@ -59,7 +62,11 @@ class Name extends React.Component {
                 <Label for="name">Full Name</Label>
```
Comment (robrap) [inline src/components/ProfilePage/Name.jsx]: Missing: Full Name

File: src/i18n/messages/es.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
   "profile.no.certificates": "Aún no tienes ningún certificado.",
-  "profile.bio.empty": "Añada una breve biografía"
+  "profile.bio.about.me": "Sobre Mi",
+  "profile.bio.empty": "Añade una breve biografía"
```
Comment (robrap) [inline src/i18n/messages/es.json]: Do you plan on leaving this file in?  Probably fine either way.  Just a reminder to remove later if you planned on removing it.

File: src/components/ProfilePage/Country.jsx [hunk 4]
```diff
@@ -95,7 +96,15 @@ class Country extends React.Component {
...
+              <FormattedMessage
+                id="profile.country.empty"
+                defaultMessage="Add location"
+                description="instructions when the user doesn't have a location set"
+              />
+            </EmptyContent>
+          ),
           static: (
             <React.Fragment>
               <EditableItemHeader content="Location" />
```
Comment (robrap) [inline src/components/ProfilePage/Country.jsx]: Missing?: Location

File: src/components/ProfilePage/Education.jsx [hunk 5]
```diff
@@ -95,10 +100,18 @@ class Education extends React.Component {
...
+                defaultMessage="Add education"
+                description="instructions when the user doesn't have their level of education set"
+              />
+            </EmptyContent>
+          ),
           static: (
             <React.Fragment>
-              <EditableItemHeader content="Education" />
+              <EditableItemHeader content={intl.formatMessage(messages['profile.education.education'])} />
               <h5>{EDUCATION[value]}</h5>
```
Comment (robrap) [inline src/components/ProfilePage/Education.jsx]: Do we need to care about the education values?

File: src/components/ProfilePage/AgeMessage.jsx [hunk 6]
```diff
@@ -1,16 +1,30 @@
...
     <UncontrolledAlert color="info">
-      <h6>Your profile cannot be shared.</h6>
-      <p>
-        To share your profile with other edX learners,
-        you must confirm that you are over the age of 13.
-      </p>
-      <a href={accountURL}>Set your date of birth</a>
+      <FormattedMessage
+        id="profile.age.headline"
+        defaultMessage="Your profile cannot be shared."
```
Comment (robrap) [inline src/components/ProfilePage/AgeMessage.jsx]: Are we intentionally not telling them why their profile cannot be shared?

File: src/i18n/messages/es.json (hunk 3 above)
Comment (albemarle) [reply src/i18n/messages/es.json]: This is the file that drives Spanish, so yes.  :-)

File: src/components/ProfilePage/Education.jsx (hunk 5 above)
Comment (albemarle) [reply src/components/ProfilePage/Education.jsx]: Oh I missed the entire Constants directory, good catch!

File: src/components/ProfilePage/AgeMessage.jsx (hunk 6 above)
Comment (albemarle) [reply src/components/ProfilePage/AgeMessage.jsx]: Yup, that's in the next line.

Comment (robrap) [reply src/components/ProfilePage/AgeMessage.jsx]: :) Oops.  Thanks.  I didn't read the end of the details.  Note: I like "header" better than "headline" if we were to document a best practice, but no change needed.

**Pull request merged**